thinc/backends/cpu_kernels.hh

```diff
@@ -395,4 +398,20 @@ void backprop_seq2col(A* d_seqs, const A* d_cols, const L* lengths, L B, L I, L
     }
 }
 
+
+template <typename I, typename L>
+void cpu_gather_add(saxpy_ptr saxpy, float* out_bo, const float* table_to, const I* indices_bk, L T, L O, L B, L K)
+{
+    for (L b = 0; b < B; ++b) {
+        for (L k = 0; k < K; ++k) {
+            I idx = indices_bk[b * K + k];
+            if (idx > T) {
+                throw std::out_of_range("Embedding index out-of-bounds");
+            }
+            (*saxpy)(O, 1.0, table_to + idx * O, 1, out_bo + b * O, 1);
```

Comment:
We should be able to call the function pointer without dereferencing it.

Reply:
Ah yes, fixed.